R/google_drive.R

```diff
@@ -0,0 +1,195 @@
+#' API to get a list of files in Google Drive
+#' @export
+#' @param teamDriveId - in case you want to search for team drive
+#' @param path - This should be ID of the folder since searching with folder name doesn't always work as expected.
+#' @param type - object type that you want to include in your query result.
+listItemsInGoogleDrive <- function(teamDriveId = NULL, path = NULL, type =  c("csv", "tsv", "txt", "folder", "xls", "xlsx")){
+  if (!requireNamespace("googledrive")) {
+    stop("package googledrive must be installed.")
+  }
+  token = getGoogleTokenForDrive()
+  googledrive::drive_set_token(token)
+  if (!is.null(path)) {
+    path = googledrive::as_id(path)
+  }
+  # If team id is provided search documents within the team.
+  if (teamDriveId != "" && !is.null(teamDriveId)) {
+    teamDriveId = googledrive::as_id(teamDriveId)
+  }
+  # To improve performance, only get id, name, mimeType, modifiedTime, size, parents for each file.
+  googledrive::drive_ls(path = path, type = type, team_drive = teamDriveId, pageSize = 1000, fields = "files/id, files/name, files/mimeType, files/modifiedTime, files/size, files/parents, nextPageToken")
+}
+
+#' API to get a folder details in Google Drive
+#' @export
+getGoogleDriveFolderDetails <- function(teamDriveId = NULL , path = NULL) {
+  if(!requireNamespace("googledrive")) {
+    stop("package googledrive must be installed.")
+    }
+  token = getGoogleTokenForDrive()
+  googledrive::drive_set_token(token)
+  if (!is.null(path)) {
+    path = googledrive::as_id(path)
+  }
+  df <- NULL
+  #if team id is provided search documents within the team.
+  # If team id is provided search documents within the team.
+  if (teamDriveId != "" && !is.null(teamDriveId)) {
+    teamDriveId = googledrive::as_id(teamDriveId)
+  }
+  df <- googledrive::drive_get(team_drive = googledrive::as_id(teamDriveId), id = path)
```

Comment:
Since you already called as_id at line 38, I think you don't need as_id here again. Please test after this change.

Reply:
removed the duplicated as_id call.